e3sm_diags/driver/utils/regrid.py

```diff
@@ -264,6 +267,12 @@ def _apply_land_sea_mask(
 def _subset_on_region(ds: xr.Dataset, var_key: str, region: str) -> xr.Dataset:
     """Subset a variable in the dataset based on the region.
 
+    This function makes sure that the axes/axes being subsetted on is in
+    ascending order. For example, if the latitude axis is in descending order,
+    [90, 0, -90], no matches will be made on the subset if the region has a lat
+    slice spec of (-90, -55) (e.g., 'polar_S'). This is because Xarray subsets
+    in ascending order. Sorting the axis beforehand will avoid this issue.
+
```

Comment:
@chengzhuzhang Just an FYI about this fix with regional subsetting with Xarray using the region specs dictionary.

Reply:
A few ref variables were not being subsetted correctly because the lat spec for `polar_S` is `(-90, -55)` but the lat coords were in descending order (`[90, 0,... -90]`).